packages/strapi-plugin-graphql/hooks/graphql/index.js

```diff
@@ -102,11 +102,7 @@ module.exports = strapi => {
         engine: _.get(strapi.plugins.graphql, 'config.engine', false),
       };
 
-      if (
-        strapi.plugins.graphql.config.tracing ||
-        strapi.plugins.graphql.config.introspection ||
-        strapi.plugins.graphql.config.engine
-      ) {
+      if (_.has(strapi.plugins.graphql.config, ['tracing', 'introspection', 'engine'])) {
```

Comment:
Nice try but this will check if you have strapi.plugins.graphql.config.tracing.introspection.engine here :) the array is considered as a path :) you will need 3 has ;)

Reply:
Ouch! I assumed this would work, but you are totally rigth!